k4MarlinWrapper/src/components/Lcio2EDM4hep.cpp

```diff
@@ -62,12 +68,16 @@ StatusCode Lcio2EDM4hepTool::finalize() { return AlgTool::finalize(); }
 // Check if a collection was already registered to skip it
 // **********************************
 bool Lcio2EDM4hepTool::collectionExist(const std::string& collection_name) {
-  auto collections = m_podioDataSvc->getEventFrame().getAvailableCollections();
-  for (const auto& name : collections) {
-    if (collection_name == name) {
-      debug() << "Collection named " << name << " already registered, skipping conversion." << endmsg;
-      return true;
-    }
+  std::vector<std::string> collections;
+  if (m_podioDataSvc) {
+    collections = m_podioDataSvc->getEventFrame().getAvailableCollections();
+  } else {
+    std::optional<std::map<uint32_t, std::string>> dummy = std::nullopt;
+    getAvailableCollectionsFromStore(this, dummy, true);
```

Comment:
For my education: Why are we not assigning the return value here to `collections`? It looks like `collections` will always be empty if we use the `IOSvc` and the `std::find` on line 78, will never find the `collection_name`?

Reply:
I forgot to set it, now it should be fixed. This is used in the `global_converter_maps` test where `EventHeader` is skipped a few times after the first one. I see that after adding this in the output of the test it shows that is being skipped as it happens in the original test without IOSvc.